source/test/ItemTest.java

```diff
@@ -8,62 +11,71 @@ class ItemTest {
+        item1 = new Item("Fees",99999, Item.ItemType.BILL);
     }
     @Test
     void testGetId() {
-        assertEquals(0,item0.getId());
-        assertEquals(1,item1.getId());
+        assertEquals(0, item0.getId());
+        assertEquals(1, item1.getId());
     }
-
     @Test
     void testGetName() {
-        assertEquals("pepsi",item0.getName());
-        assertEquals("fees",item1.getName());
+        assertEquals("Pepsi", item0.getName());
+        assertEquals("Fees", item1.getName());
+    }
+    @Test
+    void getType() {
+        Assertions.assertEquals(Item.ItemType.ITEM, item0.getType());
+        Assertions.assertEquals(Item.ItemType.BILL, item1.getType());
     }
-
     @Test
     void testGetPrice() {
-        assertEquals(5,item0.getPrice());
-        assertEquals(99999,item1.getPrice());
+        assertEquals(5, item0.getPrice());
+        assertEquals(99999, item1.getPrice());
         assertThrows(IllegalArgumentException.class ,()->
-                new Item(Item.ItemType.ITEM,-23,"water"));
+                new Item("Water", -23, Item.ItemType.ITEM));
     }
-
-
     @Test
-    void getDatePurchased() {
+    void testBuy() {
         User user = new User("Bruce","batman","ilovecatwoman");
         Account account = new Account(user);
         account.deposit(5000);
-        item0.purchase(account);
-        assertEquals(LocalDateTime.now().truncatedTo(ChronoUnit.SECONDS),item0.getDatePurchased().truncatedTo(ChronoUnit.SECONDS));
-    }
+        assertTrue(item0.buy(account));
+        assertFalse(item1.buy(account));
 
+    }
     @Test
-    void getAccount() {
-        User user = new User("Bruce","batman","ilovecatwoman");
+    void getDatePurchased() {
+        User user = new User("Bruce", "batman", "ilovecatwoman");
         Account account = new Account(user);
         account.deposit(5000);
-        item0.purchase(account);
-        assertEquals(account,item0.getAccount());
-    }
 
+
+        assertEquals(LocalDateTime.MIN, item0.getDatePurchased());
+        assertEquals(LocalDateTime.MIN, item1.getDatePurchased());
+
+        assertTrue(item0.buy(account));
+
+        assertEquals(
+                LocalDateTime.now().truncatedTo(ChronoUnit.SECONDS),
+                item0.getDatePurchased().truncatedTo(ChronoUnit.SECONDS)
+        );
+    }
     @Test
-    void purchase() {
-        User user = new User("Bruce","batman","ilovecatwoman");
-        Account account = new Account(user);
+    void getAccount() {
+        User user1 = new User("Bruce","batman","ilovecatwoman");
+        User user2 = new User("Bruce","batman","ilovecatwoman");
+        User user3 = new User("Bruce","batman","ilovecatwoman");
+        User user4 = new User("Bruce","batman","ilovecatwoman");
+        Account account = new Account(user4);
         account.deposit(5000);
-        assertTrue(item0.purchase(account));
-        assertFalse(item1.purchase(account));
 
-    }
+        assertTrue(item0.buy(account));
```

Comment:
remove the unnecessary assertion

Reply:
Again checks for the same reason

Reply:
the testing should focus solely on the getAccountNumber functionality, and change the name of the test function to match the function being tested.

Reply:
Good point, will revert it back. But then getAccountNumber should fail automatically if buying fails right?